packages/flutter_tools/lib/src/ios/mac.dart

```diff
@@ -215,7 +215,7 @@ Future<XcodeBuildResult> buildXcodeProject({
   final bool hasFlutterPlugins = injectPlugins();
 
   if (hasFlutterPlugins)
-    await _runPodInstall(appDirectory, flutterFrameworkDir(mode));
+    await cocoaPods.processPods(appDirectory, flutterFrameworkDir(mode));
```

Comment:
I'm a bit ambivalent about this in #9827. It's to address the case where the user manually added pod dependencies but isn't using any plugins via flutter right? In those cases, I imagine the user is integrating pods to use in their native non-flutter code. I think running pod install manually in those cases more closely matches the normal iOS non-flutter development flow. pod install doesn't integrate with xcode build/run and nothing in flutter affects pods like the pubspec would if they were using plugins. 

I think we should just let the users run it themselves since it's disconnected from their flutter experience.